pkg/cloudprovider/types.go

```diff
@@ -104,11 +104,11 @@ func (its InstanceTypes) OrderByPrice(reqs scheduling.Requirements) InstanceType
 	sort.Slice(its, func(i, j int) bool {
 		iPrice := math.MaxFloat64
 		jPrice := math.MaxFloat64
-		if len(its[i].Offerings.Available().Compatible(reqs)) > 0 {
-			iPrice = its[i].Offerings.Available().Compatible(reqs).Cheapest().Price
+		if ofs := its[i].Offerings.Available().Compatible(reqs); len(ofs) > 0 {
+			iPrice = ofs.Cheapest().Price
 		}
-		if len(its[j].Offerings.Available().Compatible(reqs)) > 0 {
-			jPrice = its[j].Offerings.Available().Compatible(reqs).Cheapest().Price
+		if ofs := its[j].Offerings.Available().Compatible(reqs); len(ofs) > 0 {
+			jPrice = ofs.Cheapest().Price
```

Comment:
nice! this should be a speed up! Can you run + post scheduling benchmarks so we can see?